lib/oli_web/live/delivery/student/learn_live.ex

```diff
@@ -133,6 +137,42 @@ defmodule OliWeb.Delivery.Student.LearnLive do
     end
   end
 
+  _docp = """
+  If there is an unfinished resource (progress < 100%) at level 1 of the hierarchy, the learn page will
+  auto-scroll to that resource.
+  This resource could be a unit or a top-level page.
+  After scrolling, we set the scroll_to_first_unfinished_level_1_resource flag to false
+  to prevent auto-scrolling from being re-triggered with every user interaction.
+  """
+
+  defp maybe_scroll_to_first_unfinished_level_1_resource(
+         %{assigns: %{scroll_to_first_unfinished_level_1_resource: false}} = socket
+       ),
+       do: socket
+
+  defp maybe_scroll_to_first_unfinished_level_1_resource(socket) do
```

Comment:
I'm concerned about the complexity of the underlying query here, given that we are running it within a Phoenix LiveView event handler.  Instead of only asychronously initiating the scroll, we may want to asychronously execute the query and do the scroll so that we do not introduce notieable delays within the UI.